app/services/discussion_service.rb

```diff
@@ -55,7 +55,25 @@ def self.update(discussion:, params:, actor:)
     discussion.save!
 
     ThreadSearchService.index! discussion.id if update_search_vector
-    DiscussionReader.for(discussion: discussion, user: actor).set_volume_as_required!
+    DiscussionReader.for(user: actor, discussion: discussion).set_volume_as_required!
     event
   end
+
+  def self.update_reader(discussion:, params:, actor:)
+    reader = DiscussionReader.for(discussion: discussion, user: actor)
+    actor.ability.authorize! :show, discussion
```

Comment:
I thought this wanted to be `authorize :update, reader`, but that doesn't really work because it only checks that the reader's user is equal to the permission user.. which is inherently true here.

I could throw that check in here as well, but what I really want to be sure of is that the user can see the discussion s/he's trying to read.